### PR TITLE
book: typo fixes, wording improvements.

### PR DESCRIPTION
**Pull request by wheals**

### CODE REVIEW

File: src/doc/trpl/iterators.md [hunk 1]
```diff
@@ -212,9 +212,9 @@ see why consumers matter.
 As we've said before, an iterator is something that we can call the
 `.next()` method on repeatedly, and it gives us a sequence of things.
 Because you need to call the method, this means that iterators
-are *lazy* and don't need to generate all of the values upfront.
-This code, for example, does not actually generate the numbers
-`1-100`, and just creates a value that represents the sequence:
+can be *lazy* and not generate all of the values upfront. This code,
```
Comment (steveklabnik) [inline src/doc/trpl/iterators.md]: Currently, all iterators are lazy, but this is good future-proofing :)

**Comment by steveklabnik** [comment]
Thanks so much!

File: src/doc/trpl/primitive-types.md [hunk 2]
```diff
@@ -82,7 +82,7 @@ Let’s go over them by category:
 Integer types come in two varieties: signed and unsigned. To understand the
 difference, let’s consider a number with four bits of size. A signed, four-bit
 number would let you store numbers from `-8` to `+7`. Signed numbers use
-“two’s compliment representation”. An unsigned four bit number, since it does
+“two’s complement representation”. An unsigned four bit number, since it does
```
Comment (apasel422) [inline src/doc/trpl/primitive-types.md]: Can this be changed from `four bit` to `four-bit`?

**Pull request merged**